client/app/scripts/selectors/nodes-chart.js

```diff
@@ -0,0 +1,32 @@
+import { createSelector } from 'reselect';
+import { Map as makeMap } from 'immutable';
+
+
+const allNodesSelector = state => state.get('nodes');
+
+export const nodesSelector = createSelector(
+  [
+    allNodesSelector,
+  ],
+  allNodes => allNodes.filter(node => !node.get('filtered'))
+);
+
+
+export const nodeAdjacenciesSelector = createSelector(
```

Comment:
That's a good point. I put an extensive `TODO` in the code that explains how we could exploit new selectors for smarter layout engine caching, but one prerequisite here would be to implement *deep update* of the nodes in `reducers/root.js`, where currently we are just doing:

```
state = state.setIn(['nodes', node.id], fromJS(node))
```

A consequence of that single line is that everything about the node always gets updated, so the case e.g. of node metrics being updated but the adjacencies not being modified actually never happens.